tutorials/cloud-spanner-aspnet/index.md

```diff
@@ -0,0 +1,107 @@
+
+![image alt text](image_2.png)
+
+### **Model and Controller Class**
+
+The first code we need to write is a Model class for a Book.  Add a new folder called "Models" in the project and then a class called Book to that folder with the following content:
+
+    namespace SpannerTest.Models
+    {
+        public class Book
+        {
+            public string Id { get; set; }
+            public string Author { get; set; }
+            public string Title { get; set; }
+            public DateTime PublishDate { get; set; }
+        }
+    }
+
+We also need a Controller class to handle the http requests.  Under the "Controllers" folder add a new “Web API Controller class” item named **BooksController**.  Erase the contents of this controller.  We’ll start filling it in below.
+
+### **Inserting **
+
+This sample will be pretty basic and will not have everything you would normally see in an ASP.NET application, but does show how to use Cloud Spanner.
+
+Add the following code to BooksController. Be sure to replace {myProject} with your own project id.
+
+    namespace SpannerTest.Controllers
    {
        [Route("api/[controller]/")]
        public class BooksController : Controller
        {
            [HttpPost]
            public async Task<IActionResult> Create([FromBody] Book item)
            {
                // Insert a new item.
                using (var connection =
                    new SpannerConnection(
                        $"Data Source=projects/{_myProject}/instances/myspanner/databases/books"))
                {
                    await connection.OpenAsync();
     
                    item.Id = Guid.NewGuid().ToString("N");
                    var cmd = connection.CreateInsertCommand(
                        "bookTable", new SpannerParameterCollection
                        {
                            {"ID", SpannerDbType.String, item.Id},
                            {"Title", SpannerDbType.String, item.Title},
                            {"Author", SpannerDbType.String, item.Author},
                            {"PublishDate", SpannerDbType.Date, item.PublishDate}
                        });
     
                    await cmd.ExecuteNonQueryAsync();
                }
     
                return Ok();
            }
+        }
+    }
+
+
+When you run the application, use a utility to send different REST requests to test your API.  I like using [Postman](https://chrome.google.com/webstore/detail/postman/fhbjgbiflinjbdggehcddcbncdddomop), which is a Chrome extension but you can use something else like curl.  Use Postman to add a new book to the database as below.  Note: replace the port number to match your server.  So, the ‘56777’ in [http://localhost:56777](http://localhost:56777) should be replaced with the port your local server is running on.
```

Comment:
Suggest moving "when you run the application" paragraph to a new section Adding books and testing the application.